libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -0,0 +1,297 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libsolidity/analysis/DeclarationTypeChecker.h>
+
+#include <libsolidity/analysis/ConstantEvaluator.h>
+
+#include <libsolidity/ast/TypeProvider.h>
```

Comment:
De we need empty lines between same level include files (libsolidity)

Reply:
Not sure we have a clear convention for this, but I can check. Note that this PR is so far only an early draft, so detailed reviewing doesn't make too much sense besides regarding the general concept ;-).

Reply:
I agree, just wanted to learn some code troughout review, and these comments are some basic things I saw

Reply:
@mijovic it depends. The header that the cpp file relates to, in this case `DeclarationTypeChecker.h`, usually has a blank line after.
Then how the subsystems are separated is usually up to you, as long as it goes from specific to general (where the last ones are `std`) and it's readable.